packages/twenty-server/src/engine/core-modules/file/jobs/cleanup-orphaned-files.job.ts

```diff
@@ -0,0 +1,49 @@
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { In, Repository } from 'typeorm';
+
+import { FileEntity } from 'src/engine/core-modules/file/entities/file.entity';
+import { FileMetadataService } from 'src/engine/core-modules/file/services/file-metadata.service';
+import { Process } from 'src/engine/core-modules/message-queue/decorators/process.decorator';
+import { Processor } from 'src/engine/core-modules/message-queue/decorators/processor.decorator';
+import { MessageQueue } from 'src/engine/core-modules/message-queue/message-queue.constants';
+
+export type CleanupOrphanedFilesJobData = {
+  workspaceId: string;
+  fileIds: string[];
+};
+
+@Processor(MessageQueue.deleteCascadeQueue)
+export class CleanupOrphanedFilesJob {
+  constructor(
+    @InjectRepository(FileEntity, 'core')
+    private readonly fileRepository: Repository<FileEntity>,
+    private readonly fileMetadataService: FileMetadataService,
+  ) {}
+
+  @Process(CleanupOrphanedFilesJob.name)
```

Comment:
I don't see that much benefit into splitting this in it's own job since it's already an asynchronous process, I feel it adds a bit of unneccesary complexity but why not...